packages/webapp/src/containers/Animals/AddAnimals/utils.ts

```diff
@@ -28,15 +23,146 @@ import {
   DefaultAnimalBreed,
   DefaultAnimalType,
 } from '../../../store/api/types';
+import { toLocalISOString } from '../../../util/moment';
+import { DetailsFields, type AnimalDetailsFormFields } from './types';
+import {
+  AnimalSummary,
+  BatchSummary,
+} from '../../../components/Animals/AddAnimalsSummaryCard/types';
 import { chooseAnimalBreedLabel, chooseAnimalTypeLabel } from '../Inventory/useAnimalInventory';
 
-// TODO
-export const formatAnimalDetailsToDBStructure = (data: any) => {
-  return data;
+const formatFormTypeOrBreed = (
+  typeOrBreed: 'type' | 'breed',
+  data?: { label: string; value: string; __isNew__?: boolean },
```

Comment:
I think it might be the proper way -- it's even in one code example on the [React Select website](https://react-select.com/advanced#custom-filter-logic) -- it's just so weird they named it like that and aren't describing it in the API docs!

Reply:
Oh that's a relief! Thank you for finding that!